jvm/src/test/scala/SymbolSuite.scala

```diff
@@ -48,93 +37,134 @@ class SymbolSuite extends BaseUnpicklingSuite {
       s"Condition does not hold for ${getDeclsByPrefix(ctx, prefix).filter(!condition(_))}"
     )
 
-  def assertContainsExactly(ctx: FileContext, prefix: DeclarationPath, symbolNames: Set[Name]): Unit = {
-    val decls = getDeclsByPrefix(ctx, prefix).map(_.name)
+  def assertContainsExactly(ctx: FileContext, prefix: DeclarationPath, symbolPaths: Set[DeclarationPath]): Unit = {
+    val decls = getDeclsByPrefix(ctx, prefix)
+    val expected = symbolPaths.toList.map(p => resolve(p)(using ctx))
     // each declaration is in the passed set
     assert(
-      decls.forall(symbolNames.contains(_)),
-      s"Unexpected declarations: ${decls.filter(!symbolNames.contains(_)).toDebugString}"
+      decls.forall(decls.contains(_)),
+      s"Unexpected declarations: ${decls.filter(!expected.contains(_)).map(_.name).toDebugString}"
     )
     // every name in the passed set is a declaration
     assert(
-      symbolNames.forall(decls.contains(_)),
-      s"Declaration not found: ${symbolNames.filter(!decls.contains(_)).toDebugString}"
+      expected.forall(decls.contains(_)),
+      s"Declaration not found: ${expected.filter(!decls.contains(_)).map(_.name).toDebugString}"
     )
   }
 
   test("basic-symbol-structure") {
-    val ctx = getUnpicklingContext("empty_class/EmptyClass")
-    assertContainsDeclaration(ctx, SimpleName("empty_class") :: TypeName(SimpleName("EmptyClass")) :: Nil)
+    val EmptyClass = empty_class / tname"EmptyClass"
+    val ctx = getUnpicklingContext(EmptyClass)
+    assertContainsDeclaration(ctx, EmptyClass)
     // EmptyClass and its constructor are the only declarations in empty_class package
     assertContainsExactly(
       ctx,
-      SimpleName("empty_class") :: Nil,
-      Set(SimpleName("empty_class"), TypeName(SimpleName("EmptyClass")), SimpleName("<init>"))
+      empty_class,
+      Set(empty_class / tname"EmptyClass", empty_class / tname"EmptyClass" / name"<init>")
     )
   }
 
-  test("inner-class") {
-    val ctx = getUnpicklingContext("simple_trees/InnerClass")
-    // Inner is a declaration in InnerClass
-    assertContainsDeclaration(
+  test("basic-symbol-structure-symbollic-names".ignore) {
```

Comment:
```suggestion
  test("basic-symbol-structure-symbolic-names".ignore) {
```

Reply:
And the same everywhere else. "symbollic" is a common misspelling.

Reply:
Todo: make API's private